libs/libvtrutil/src/vtr_thread_pool.h

```diff
@@ -0,0 +1,171 @@
+#pragma once
+
+/** 
+ * @file RouterThreadPool.h
+ * @brief A generic thread pool for parallel task execution
+ */
+
+#include <thread>
+#include <queue>
+#include <mutex>
+#include <condition_variable>
+#include <memory>
+#include <atomic>
+#include <functional>
+#include <cstddef>
+#include <vector>
+#include "vtr_log.h"
+#include "vtr_time.h"
+
+namespace vtr {
+
+class thread_pool {
+private:
+    struct ThreadData {
+        std::thread thread;
+        std::queue<std::function<void()>> task_queue;
+        std::mutex queue_mutex;
+        std::condition_variable cv;
```

Comment:
Shouldn't the name give an idea of what the condition variable controls/is used for, rather than just cv?